timesketch/lib/analyzers/sequence_sessionizer_test.py

```diff
@@ -0,0 +1,430 @@
+import copy
+
+import mock
+
+from timesketch.lib.analyzers.sequence_sessionizer \
+import SequenceSessionizerSketchPlugin
+from timesketch.lib.testlib import BaseTest
+from timesketch.lib.testlib import MockDataStore
+from timesketch.lib.analyzers.interface import Event
+from timesketch.models.user import User
+from timesketch.models.sketch import Sketch
+
+
+class ManyEventsSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    session_type = 'mock_seq_sessionizer'
+    max_time_diff_micros = 100
+    return_fields = ['hostname', 'source_short', 'timestamp']
+    event_seq = [{
+        'hostname': 'host',
+        'source_short': 'FILE'
+    }, {
+        'hostname': 'host',
+        'source_short': 'WEBHIST'
+    }]
+
+
+class OneEventSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    max_time_diff_micros = 100
+    session_type = 'one_event_seq_sessionizer'
+    event_seq = [{'hostname': 'host', 'source_short': 'FILE'}]
+    return_fields = ['hostname', 'source_short', 'timestamp']
+
+
+# Invalid sequence sessionizers.
+class NoneSessionTypeSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    session_type = 'name'
+    event_seq = [{'hostname': 'host', 'source_short': 'FILE'}]
+    return_fields = ['timestamp', 'source_short']
+
+
+class NoneSeqSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    session_type = 'valid_name'
+    return_fields = ['timestamp', 'source_short']
+
+
+class EmptySeqSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    event_seq = []
+    session_type = 'valid_name'
+    return_fields = ['timestamp', 'source_short']
+
+
+class NoTimestampSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    session_type = 'valid_name'
+    event_seq = [{'hostname': 'host', 'source_short': 'FILE'}]
+    return_fields = ['source_short']
+
+
+class NoReturnAttrSequenceSessionizer(SequenceSessionizerSketchPlugin):
+    session_type = 'valid_name'
+    event_seq = [{'hostname': 'host', 'source_short': 'FILE'}]
+    return_fields = ['timestamp']
+
+
+# List of dictionaries with attributes for mock events.
+one_session_args = [{
+    'hostname': 'host',
+    'source_short': 'FILE'
+}, {
+    'hostname': 'host',
+    'source_short': 'WEBHIST'
+}]
+
+two_sessions_args = [{
+    'hostname': 'host',
+    'source_short': 'FILE'
+}, {
+    'hostname': 'host',
+    'source_short': 'WEBHIST'
+}, {
+    'hostname': 'host',
+    'source_short': 'FILE'
+}, {
+    'hostname': 'host',
+    'source_short': 'WEBHIST'
+}]
+
+one_event_seq_args = [{'hostname': 'host', 'source_short': 'FILE'}]
+
+
+class TestSequenceSessionizerPlugin(BaseTest):
+    """Tests the functionality of the sequence sessionizing sketch analyzer."""
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_session_type_none(self):
+        """Test session_type is not None."""
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
```

Comment:
It's better to write it this way:

```python
self.assertRaises(
    RuntimeError, 
    sessionizer=NoneSessionTypeSequenceSessionizer(index, sketch_id))
```

Same for the other assertions in the tests.